components/ipfs/ipfs_import_worker_base.cc

```diff
@@ -0,0 +1,303 @@
+
+#include <utility>
+
+#include "base/command_line.h"
+#include "base/files/file_util.h"
+#include "base/guid.h"
+#include "base/strings/strcat.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "base/threading/scoped_blocking_call.h"
+#include "base/time/time.h"
+#include "brave/components/ipfs/ipfs_constants.h"
+#include "brave/components/ipfs/ipfs_json_parser.h"
+#include "brave/components/ipfs/ipfs_utils.h"
+#include "brave/components/ipfs/pref_names.h"
+#include "brave/components/ipfs/service_sandbox_type.h"
+#include "components/grit/brave_components_strings.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/service_process_host.h"
+#include "content/public/browser/storage_partition.h"
+#include "net/base/mime_util.h"
+#include "net/base/url_util.h"
+#include "net/http/http_request_headers.h"
+#include "net/http/http_response_headers.h"
+#include "net/http/http_status_code.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "storage/browser/blob/blob_data_builder.h"
+#include "storage/browser/blob/blob_data_handle.h"
+#include "storage/browser/blob/blob_impl.h"
+#include "storage/browser/blob/blob_storage_context.h"
+#include "third_party/blink/public/mojom/blob/serialized_blob.mojom.h"
+#include "url/gurl.h"
+
+namespace {
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("ipfs_import_worker", R"(
+          semantics {
+            sender: "IPFS import worker"
+            description:
+              "This worker is used to communicate with IPFS daemon "
+            trigger:
+              "Triggered by user actions"
+            data:
+              "Options of the commands."
+            destination: WEBSITE
+          }
+          policy {
+            cookies_allowed: NO
+            setting:
+              "You can enable or disable this feature in brave://settings."
+            policy_exception_justification:
+              "Not implemented."
+          }
+        )");
+}
+
+// Return a date string formatted as "YYYY-MM-DD".
+std::string TimeFormatDate(const base::Time& time) {
+  base::Time::Exploded exploded_time;
+  time.UTCExplode(&exploded_time);
+  return base::StringPrintf("%04d-%02d-%02d", exploded_time.year,
+                            exploded_time.month, exploded_time.day_of_month);
+}
+
+}  // namespace
+
+namespace ipfs {
+
+IpfsImportWorkerBase::IpfsImportWorkerBase(content::BrowserContext* context,
+                                           const GURL& endpoint,
```

Comment:
Maybe add some error handling for context being null and endpoint being invalid or empty?